src/CmakeRegen.cpp

```diff
@@ -44,8 +46,8 @@ void RegenerateCmakeFilesForComponent(Component *comp, bool dryRun) {
         std::set<std::string> publicDeps, privateDeps, publicIncl, privateIncl;
         std::list<std::string> files;
         for (auto &fp : comp->files) {
-            files.push_back(fp->path.generic_string().c_str() + compname.size() + 3);
-            boost::filesystem::path p = fp->path;
+            files.push_back(fp->path.string().c_str() + compname.size() + 3);
```

Comment:
This reintroduces a bug on Windows where it now will use backslashes, and that breaks other logic. We need generic_string here.